### PR TITLE
Tominabox1 userspace creation

### PR DESCRIPTION
**Pull request by tominabox1**
## Description
Created tominabox1 userspace and removed extra files from my keymaps in Dimple, CRKBD, and Minivan keyboards. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -73,3 +73,12 @@ id_rsa_*
 
 # python things
 __pycache__
+
```
Comment (drashna) [inline .gitignore]: Could you remove this change? 

You can actually add a .gitignore file on these folders, and have it ignore just those files.

File: keyboards/crkbd/keymaps/drashna/keymap.c [hunk 2]
```diff
@@ -131,9 +131,9 @@ bool process_record_keymap(uint16_t keycode, keyrecord_t *record) {
 oled_rotation_t oled_init_user(oled_rotation_t rotation) { return OLED_ROTATION_270; }
 
 void add_keylog(uint16_t keycode) {
-    if ((keycode >= QK_MOD_TAP && keycode <= QK_MOD_TAP_MAX) || (keycode >= QK_LAYER_TAP && keycode <= QK_LAYER_TAP_MAX)) {
-        keycode = keycode & 0xFF;
-    }
+//    if ((keycode >= QK_MOD_TAP && keycode <= QK_MOD_TAP_MAX) || (keycode >= QK_LAYER_TAP && keycode <= QK_LAYER_TAP_MAX)) {
```
Comment (drashna) [inline keyboards/crkbd/keymaps/drashna/keymap.c]: Could you revert this change to my keymap?

File: keyboards/hhkb/keymaps/tominabox1/config.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        UBRR1H = ((uint16_t)SERIAL_UART_UBRR>>8);  /* baud rate */ \
+        UCSR1B |= (1<<RXCIE1) | (1<<RXEN1); /* RX interrupt, RX: enable */ \
+        UCSR1B |= (0<<TXCIE1) | (1<<TXEN1); /* TX interrupt, TX: enable */ \
+        UCSR1C |= (0<<UPM11) | (0<<UPM10);  /* parity: none(00), even(01), odd(11) */ \
+        UCSR1D |= (0<<RTSEN) | (0<<CTSEN);  /* RTS, CTS(no flow control by hardware) */ \
+        DDRD |= (1<<5); PORTD &= ~(1<<5);   /* RTS for flow control by firmware */ \
+        sei(); \
+    } while(0)
+    #define SERIAL_UART_RTS_LO()    do { PORTD &= ~(1<<5); } while (0)
+    #define SERIAL_UART_RTS_HI()    do { PORTD |=  (1<<5); } while (0)
```
Comment (drashna) [inline keyboards/hhkb/keymaps/tominabox1/config.h]: Pretty much this entire file could be removed

File: keyboards/hhkb/keymaps/tominabox1/hhkb_teensy_avr.h [hunk 4]
```diff
@@ -0,0 +1,59 @@
+#ifndef HHKB_AVR_H
```
Comment (drashna) [inline keyboards/hhkb/keymaps/tominabox1/hhkb_teensy_avr.h]: Is this actually needed?

File: keyboards/hhkb/keymaps/tominabox1/matrix.c [hunk 5]
```diff
@@ -0,0 +1,216 @@
+/*
```
Comment (drashna) [inline keyboards/hhkb/keymaps/tominabox1/matrix.c]: Is this actually needed?

File: keyboards/hhkb/keymaps/tominabox1/rules.mk [hunk 6]
```diff
@@ -0,0 +1,109 @@
+
```
Comment (fauxpark) [inline keyboards/hhkb/keymaps/tominabox1/rules.mk]: Please delete this file, it's exactly the same as the keyboard-level rules.mk.

File: keyboards/crkbd/keymaps/drashna/keymap.c (hunk 2 above)
Comment (tominabox1) [reply keyboards/crkbd/keymaps/drashna/keymap.c]: Honestly don't even know how this made it in! my bad. will revert

File: users/tominabox1/tominabox1.h [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    RGBRST,
+    KC_PASTA
+};
+
+enum tap_dance_indexes {
+    KC_EMAIL,
+    TD_SFT_CPS,
+};
+
+#define KC______ KC_TRNS
```
Comment (fauxpark) [inline users/tominabox1/tominabox1.h]:
```suggestion
```

File: users/tominabox1/tominabox1.h [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+#define KC_XXXXX KC_NO
```
Comment (fauxpark) [inline users/tominabox1/tominabox1.h]:
```suggestion
```

File: users/tominabox1/tominabox1.h [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+#define KC_ARROW MO(4)
+#define KC_RST   RESET
```
Comment (fauxpark) [inline users/tominabox1/tominabox1.h]:
```suggestion
```

File: keyboards/hhkb/keymaps/tominabox1/.gitignore [hunk 10]
```diff
@@ -0,0 +1,3 @@
+*.h
+matrix.c
+rules.mk
```
Comment (fauxpark) [inline keyboards/hhkb/keymaps/tominabox1/.gitignore]: This file should be removed.

**Comment by fauxpark** [comment]
There are some permissions changes here. Run this:
```
git update-index --chmod=-x path/to/file
```

on each of these files:
```
keyboards/crkbd/keymaps/drashna/keymap.c
keyboards/lazydesigners/dimple/keymaps/default/keymap.c
keyboards/lazydesigners/dimple/keymaps/default/readme.md
```

and then commit & push.

File: keyboards/hhkb/keymaps/tominabox1/.gitignore (hunk 10 above)
Comment (tominabox1) [reply keyboards/hhkb/keymaps/tominabox1/.gitignore]: So I was using TMK years ago with a Teensy so I just copied some stuff over so that it worked. I'm not sure what is being used in Teensy. I'd like to make sure this works before pushing to the main, so thought I'd just hide them for now.

File: .gitignore (hunk 1 above)
Comment (tominabox1) [reply .gitignore]: moved to my user folder

File: keyboards/crkbd/keymaps/drashna/keymap.c (hunk 2 above)
Comment (tominabox1) [reply keyboards/crkbd/keymaps/drashna/keymap.c]: should be reverted to your master

File: users/tominabox1/rules.mk [hunk 11]
```diff
@@ -0,0 +1,27 @@
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
+#
+CONSOLE_ENABLE = no
+TAP_DANCE_ENABLE = yes
+NKRO_ENABLE = yes
+BOOTMAGIC_ENABLE = full      # Virtual DIP switch configuration(+1000)
```
Comment (fauxpark) [inline users/tominabox1/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = full      # Virtual DIP switch configuration
```

File: users/tominabox1/rules.mk [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+ifeq ($(strip $(KEYBOARD)), crkbd/rev1)
+RGB_MATRIX_ENABLE = WS2812
+EXTRAFLAGS += -flto
+BOOTLOADER = qmk-dfu
+OLED_DRIVER_ENABLE = yes
+BOOTMAGIC_ENABLE = no
+endif
+
+ifeq ($(strip $(KEYBOARD)), lazydesigners/dimple)
+MOUSEKEY_ENABLE = no        # Mouse keys(+4700)
```
Comment (fauxpark) [inline users/tominabox1/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = no        # Mouse keys
```

File: users/tominabox1/rules.mk [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+EXTRAKEY_ENABLE = yes        # Audio control and System control(+450)
```
Comment (fauxpark) [inline users/tominabox1/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = yes        # Audio control and System control
```
